hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java

```diff
@@ -279,4 +284,53 @@ private boolean isEmpty(Collection c) {
         return c == null || c.isEmpty();
     }
 
+    @Override
+    public void buildExport(OutputStream os) throws IOException {
+        try (Writer writer = new BufferedWriter(new OutputStreamWriter(os))) {
+            writer.write("{\"types\":[\n");
+            int offset = 0;
+            boolean hasMore = true;
+            boolean first = true;
+            while (hasMore) {
+                List<ResourceType> batch = qResourceType.from(ResourceType.class)
+                        .maxResults(MAX_RESULTS)
+                        .startOffset(offset)
+                        .build()
+                        .list();
+                for (ResourceType rt : batch) {
+                    if (!first) {
+                        writer.write(",\n");
+                    } else {
+                        first = false;
+                    }
+                    writer.write(JsonUtil.toJson(rt) + "\n");
+                }
+                hasMore = batch.size() == MAX_RESULTS;
+                offset += MAX_RESULTS;
+            }
+            writer.write("],\"resources\":[\n");
+            hasMore = true;
+            first = true;
+            offset = 0;
+            while (hasMore) {
+                List<Resource> batch = qResource.from(Resource.class)
+                        .maxResults(MAX_RESULTS)
+                        .startOffset(offset)
+                        .build()
+                        .list();
+                for (Resource r : batch) {
+                    if (!first) {
+                        writer.write(",\n");
+                    } else {
+                        first = false;
+                    }
+                    writer.write(JsonUtil.toJson(r) + "\n");
+                }
+                hasMore = batch.size() == MAX_RESULTS;
+                offset += MAX_RESULTS;
+            }
+            writer.write("]}\n");
+            writer.flush();
```

Comment:
I think here the response is flushed at the end, I think it should be flushed also in intermediate steps too.
As commented, I would need to run some tests to see if I am right or I miss something in my comment.

Reply:
I've updated with JsonGenerator, adding intermediate flush, the tests are still OK